packages/twenty-front/src/modules/settings/data-model/fields/forms/date/components/SettingsDataModelFieldDateForm.tsx

```diff
@@ -30,27 +59,74 @@ export const SettingsDataModelFieldDateForm = ({
 }: SettingsDataModelFieldDateFormProps) => {
   const { t } = useLingui();
 
-  const { control } = useFormContext<SettingsDataModelFieldDateFormValues>();
+  const { control, watch } =
+    useFormContext<SettingsDataModelFieldDateFormValues>();
 
-  const { initialDisplayAsRelativeDateValue } =
+  const { initialDisplayFormat, initialCustomUnicode35DateFormat } =
     useDateSettingsFormInitialValues({
       fieldMetadataItem,
     });
 
+  const displayFormatFromForm = watch('settings.displayFormat');
+
+  const showCustomFormatTextInput = displayFormatFromForm
+    ? isDateFieldCustomDisplayFormat(displayFormatFromForm)
+    : isDateFieldCustomDisplayFormat(initialDisplayFormat);
+
   return (
-    <Controller
-      name="settings.displayAsRelativeDate"
-      control={control}
-      defaultValue={initialDisplayAsRelativeDateValue}
-      render={({ field: { onChange, value } }) => (
-        <SettingsOptionCardContentToggle
-          Icon={IconSlash}
-          title={t`Display as relative date`}
-          checked={value ?? false}
-          disabled={disabled}
-          onChange={onChange}
+    <>
+      <Controller
+        name="settings.displayFormat"
+        control={control}
+        defaultValue={initialDisplayFormat}
+        render={({ field: { onChange, value } }) => (
+          <SettingsOptionCardContentSelect
+            Icon={IconSlash}
+            title={t`Display Format`}
+            disabled={disabled}
+            description={t`Choose the format used to display date value`}
+          >
+            <Select<FieldDateDisplayFormat>
+              disabled={disabled}
+              selectSizeVariant="small"
+              dropdownWidth={120}
+              dropdownId="selectFieldDateDisplayFormat"
+              value={value}
+              onChange={onChange}
+              options={typedKeys(displayFormatsToLabelMap).map((key) => {
+                const label = displayFormatsToLabelMap[key];
+                return {
+                  label: t`${label}`,
```

Comment:
This won't work for translations. If you want to store translations in a sepate you might need to usg "msg", you can look at other places in the frontend how it's done already